src/js/meteoData.js

```diff
@@ -0,0 +1,169 @@
+      currentWeatherObject.weather[0].icon, currentWeatherObject.wind.speed, currentWeatherObject.main.pressure, currentWeatherObject.main.humidity);
+
+    resultArray.push(meteo);
+    const dividedArray = divideArray(forecastWeatherObject.list);
+    for(let i = 1; i < dividedArray.length; i++){
+        resultArray.push(transformToObject(dividedArray[i]));
+    }
+
+    return resultArray;
+  }
+
+  //funkcja pobiera dane z api i zamienia w obiekt 
+
+  async function toDataObject(fetchLink){
+    const fetchedData = await fetch(fetchLink); //pobieranie danych z API
+    const stringData = await fetchedData.text(); //konwersja pobranych danych w obiekt zapisany w postaci stringa
+    const dataObject = JSON.parse(stringData); //konwersja obiektu zapisanego w postaci stringa w prawdziwy obiekt
+  
+    return dataObject;
+  }
+
+  //todayDate zwraca obecna date
+
+  function todayDate(){ 
+    const today = new Date();
+    const date = today.getFullYear() + '-' + (today.getMonth()+1) + '-' + today.getDate();
+    const time = today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
+    const dateTime = date + ' ' + time;
+    return dateTime;
+  }
+  
+  /* 
+    Funkcja divideArray służy mi do tego aby podzielić 40 elementową tablice zawierającą prognozy pogody co 3 godziny na 6 elementową, w której każdy element
+    reprezentuje zbiór tablic prognoz odnoszących się do 1 dnia, ułatwia to póżniejsza przemiane tych tablic na dane odnośnie prognoz na dany dzień
+  */
+  function divideArray(meteoArr){
+    let dayArray = [];
+    const dividedArray = [];
+  
+    let data1 = new Date(meteoArr[0].dt_txt);
+    data1 = data1.getDay();
+    dayArray.push(meteoArr[0])
+
+    for(let i = 1; i < meteoArr.length; i++){//for w tym miejscu ponieważ zaczynam od drugiego elementu, a z tego co kojarzę wszystkie iteratory obejmują wszystkie elementy
+        let data2 = new Date(meteoArr[i].dt_txt);
+        data2 = data2.getDay();
+        if(data1 == data2){
+            dayArray.push(meteoArr[i])
+        }
+        else {
+            dividedArray.push(dayArray);
+            dayArray = [];
+            dayArray.push(meteoArr[i]);
+        }
+        data1 = data2;
+  
+    }
+
+    dividedArray.push(dayArray);
+    return dividedArray;
+  }
+
+  //transformToObject pzyjmuje jako parametr tablice zawierajaca dane prognoz pogody na dany dzien(co 3 godziny) i zamienia w obiekt (meteoObject) 
+
+  function transformToObject(dayArray){
+    const date = dayArray[0].dt_txt;
+    let temp_min = dayArray[0].main.temp_min;
+    let temp_max = dayArray[0].main.temp_max;
+    const weatherArray = [dayArray[0].weather[0].main];
+    const descriptionArray = [dayArray[0].weather[0].description];
+    const iconArray = [dayArray[0].weather[0].icon];
+    let sumTemp = 0;
+    let sumHumidity = 0;
+    let sumPressure = 0;
+    let sumWindSpeed = 0;
+
+    for(let i = 1; i < dayArray.length; i++){
```

Comment:
dlaszego tak?

Reply:
temp_min i temp_max potrzebują inicjalizacji (głównie temp_min) dlatego ustawiam ich wartość na ta pierwszego elementu i póżniej porównuje, mógłbym to podzielić na dwie pętle ale uważam że byłoby to mniej optymalne rozwiązanie skoro można zrobić to jedną.

Reply:
czemu nie mozesz ich zainicjalizowac zerami i puscic normalna petle po tym? :eyes: